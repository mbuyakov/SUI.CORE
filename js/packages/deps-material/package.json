{
  "name": "@sui/deps-material",
  "version": "0.0.1",
  "type": "commonjs",
  "dependencies": {
    "@emotion/react": "11.10.5",
    "@emotion/styled": "11.10.5",
    "@mui/icons-material": "5.14.3",
    "@mui/material": "5.14.4",
    "material-ui-popup-state": "5.0.9"
  }
}
