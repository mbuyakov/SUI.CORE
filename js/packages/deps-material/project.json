{
  "name": "deps-material",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "packages/deps-material/src",
  "projectType": "library",
  "targets": {
    "build": {
      "executor": "@nx/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/packages/deps-material",
        "main": "packages/deps-material/src/index.ts",
        "tsConfig": "packages/deps-material/tsconfig.lib.json",
        "assets": ["packages/deps-material/*.md"],
        "updateBuildableProjectDepsInPackageJson": true
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "node tools/scripts/publish.mjs deps-material {args.ver} {args.tag}"
      },
      "dependsOn": ["build"]
    },
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["packages/deps-material/**/*.ts"]
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "packages/deps-material/jest.config.ts",
        "passWithNoTests": true
      }
    }
  },
  "tags": ["type:deps"]
}
