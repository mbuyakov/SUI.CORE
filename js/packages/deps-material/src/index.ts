// Need to correct indexed suggestion in IDEA
export * from '@mui/material';
export * as MuiIcons from '@mui/icons-material';
export * as MuiColors from "@mui/material/colors";
