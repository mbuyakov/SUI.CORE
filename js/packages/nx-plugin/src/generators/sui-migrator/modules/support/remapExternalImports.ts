export const oldToNewDeps = {
  "@material-ui/icons": "@mui/icons-material",
  "@material-ui/core": "@sui/deps-material",
  "@mui/material": "@sui/deps-material",
  "material-ui-popup-state": "@sui/deps-material",
  "antd": "@sui/deps-antd",
  "typescript-ioc": "@sui/deps-ioc",
  "@sui/amcharts": "@sui/deps-amcharts",
  "react-router": "@sui/deps-router",
  "react-router-dom": "@sui/deps-router",
};
