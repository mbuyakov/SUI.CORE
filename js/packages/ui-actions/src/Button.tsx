import React, {MouseEvent} from "react";
import {Button as MuiButton, ButtonProps as MuiButtonProps} from "@sui/deps-material";
import {ProgressIcon, TooltipWrapper} from "@sui/ui-material";
import {PopconfirmSettings, useOnClick, usePopconfirm} from "./hooks";

export type ButtonProps = Omit<MuiButtonProps, 'onClick'> & {
  tooltip?: string;
  loading?: boolean;
  popconfirmSettings?: PopconfirmSettings;
  onClick: (event: MouseEvent<HTMLButtonElement>) => void | Promise<void>;
}
export const Button: React.FC<ButtonProps> =
  (
    {
      popconfirmSettings,
      tooltip,
      onClick: propsOnClick,
      loading: propsLoading,
      ...rest
    }) => {
    const popconfirm = usePopconfirm(popconfirmSettings);
    const {loading: onClickLoading, onClick} = useOnClick<MouseEvent<HTMLButtonElement>>(propsOnClick, {popconfirm});

    const loading = onClickLoading || propsLoading;

    return popconfirm.wrapper(
      <TooltipWrapper
        title={tooltip}
      >
        <MuiButton
          {...rest}
          onClick={onClick}
          startIcon={loading ? <ProgressIcon type="icon" size={rest.size}/> : rest.startIcon}
          disabled={loading || rest.disabled}
        />
      </TooltipWrapper>
    );
  }
