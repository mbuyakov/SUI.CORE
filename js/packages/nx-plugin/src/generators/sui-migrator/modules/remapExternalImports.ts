import {factory, ImportDeclaration, StringLiteral} from "typescript";
import {logWithPrefix} from "../../../utils/logger";
import {astReplace, printNode} from "@sui/lib-typescript-ast";
import {oldToNewDeps} from "./support/remapExternalImports";

const oldToNewDepsEntries = Object.entries(oldToNewDeps);
export function remapExternalImports(projectName: string, content: string): string {

  return astReplace(content, "ImportDeclaration:has(ImportClause)", (node: ImportDeclaration) => {
    const currentDepName = (node.moduleSpecifier as StringLiteral).text;
    const newDepName = oldToNewDepsEntries
      .find(([oldDepName]) => currentDepName == oldDepName || currentDepName.startsWith(oldDepName + "/"))
      ?.[1];

    if (
      !newDepName
      // Don't replace yourself
      || newDepName === projectName
    ) {
      return;
    }

    logWithPrefix("remapExternalImports", `Replace ${currentDepName} to ${newDepName}`);

    let namedImports = node.importClause.namedBindings;

    // If used default import - probably it's from submodule, replace with named import
    if (!namedImports) {
      namedImports = factory.createNamedImports([
        factory.createImportSpecifier(node.importClause.isTypeOnly, undefined, node.importClause.name)
      ]);
    }
    return printNode(
      factory.updateImportDeclaration(
        node,
        node.modifiers,
        factory.updateImportClause(
          node.importClause,
          node.importClause.isTypeOnly,
          undefined,
          namedImports
        ),
        factory.createStringLiteral(newDepName),
        node.assertClause
      )
    );
  });
}
