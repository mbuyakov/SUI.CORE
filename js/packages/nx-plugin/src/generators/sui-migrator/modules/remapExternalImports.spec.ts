import {remapExternalImports} from "./remapExternalImports";

describe("remap external imports", () => {

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should leave as is", () => {
    const consoleLog = jest.spyOn(console, "log");
    const source = `
import {a} from "@sui/deps-material";
import {b} from "antd";
    `.trim();
    expect(remapExternalImports("@sui/deps-antd",source)).toBe(source);
    expect(consoleLog).not.toBeCalled();
  });

  it("should replace default import from subpath", () => {
    const consoleLog = jest.spyOn(console, "log");
    expect(remapExternalImports("@sui/deps-antd", `
import Button from "@mui/material/Button";
    `.trim())).toMatchSnapshot();
    expect(consoleLog).toBeCalledTimes(1);
  });

  it("should replace named import", () => {
    const consoleLog = jest.spyOn(console, "log");
    expect(remapExternalImports("@sui/deps-antd",`
import {Button} from "@mui/material";
    `.trim())).toMatchSnapshot();
    expect(consoleLog).toBeCalledTimes(1);
  });
});
