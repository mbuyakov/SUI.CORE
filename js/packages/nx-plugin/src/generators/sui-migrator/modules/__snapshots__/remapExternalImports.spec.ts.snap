// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remap external imports should replace default import from subpath 1`] = `"import {Button} from \\"@sui/deps-material\\";"`;

exports[`remap external imports should replace named import 1`] = `"import {Button} from \\"@sui/deps-material\\";"`;
